rspec-core/lib/rspec/core/option_parser.rb

```diff
@@ -178,11 +178,8 @@ def parser(options)
         end
 
         parser.on('-w', '--warnings', 'Enable ruby warnings') do
-          if Object.const_defined?(:Warning) && Warning.respond_to?(:[]=)
-            # :nocov: on older Ruby without Warning
-            Warning[:deprecated] = true
-            # :nocov:
-          end
+          options[:warnings] = :all
+          Warning[:deprecated] = true
           $VERBOSE = true
```

Comment:
I fought the temptation to remove this and solely rely on `options[:warnings] = :all`, but there's [an example](https://github.com/rspec/rspec/blob/e4a549d062d20ad5ef5c98c0b28fdc613d200287/rspec-core/spec/rspec/core/option_parser_spec.rb#L392) named "      immediately enables warnings so that **warnings are issued for files loaded by `--require`**". That sounds like this option would be able to turn on warnings on early loading stage, before the `Configuration` is set up and modifies `$VERBOSE`/`Warning[:deprecated]`.

If it is so, this code needs a better explanation.